front-end/src/store/analyzeTranscriptSlice.js

```diff
@@ -31,9 +31,35 @@ export const analyzeTranscriptSlice = createSlice({
     setProjectIdToBeDisplayed: (state, action) => {
       state.projectIdToBeDisplayed = action.payload;
     },
+    toggleStar: (state, action) => {
+      // get the index of the first intent being displayed
+      const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
+      // get the transcript that is being displayed and has the indices of the intents
+      const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
+      // change the star state of the intent being changed
+      for (let i = firstIndex; i < firstIndex+3; i++) {
+        if (transcript[i].question === action.payload) {
+          transcript[i].star = !transcript[i].star;
+          break; // exit for-loop early when intent is found
+        }
+      }
+    },
+    toggleFlag: (state, action) => {
+      // get the index of the first intent being displayed
+      const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
+      // get the transcript that is being displayed and has the indices of the intents
+      const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
+      // change the bookmark state of the intent being changed
+      for (let i = firstIndex; i < firstIndex+3; i++) {
+        if (transcript[i].question === action.payload) {
+          transcript[i].flag = !transcript[i].flag;
+          break; // exit for-loop early when intent is found
```

Comment:
for future reference, there may be an easier way to do this where we instead dispatch the intent object itself from the intentdiagram, and then we can lookup the index of that intent object in analyzedTranscripts[projectIdToBeDisplayed]. Will discuss more in person tmr.

But this works fine for this ticket!